File: src/main/java/shtykh/roomplanner/service/RoomPlanner.java
package shtykh.roomplanner.service;

import shtykh.roomplanner.model.RoomLevel;
import shtykh.roomplanner.model.RoomPlan;

import java.util.List;
import java.util.Map;

public interface RoomPlanner {

    RoomPlan plan(List<Integer> roomRequest);

    void setAvailability(Map<RoomLevel, Integer> availabilities);
}

File: src/main/kotlin/shtykh/roomplanner/service/RoomPlanner.kt
package shtykh.roomplanner.service

import shtykh.roomplanner.model.RoomLevel
import shtykh.roomplanner.model.RoomPlan

interface RoomPlanner {

    fun plan(roomRequest: List<Int>): RoomPlan

    fun setAvailability(availabilities: Map<RoomLevel, Int>)
}
